lib/new_relic/agent/serverless_handler.rb

```diff
@@ -129,14 +144,92 @@ def payload_v1
         ::JSON.dump(array)
       end
 
-      def payload_v2
+      def payload_v2 # New Relic serverless payload v2
         json = NewRelic::Agent.agent.service.marshaller.dump(@payloads)
         gzipped = NewRelic::Agent::NewRelicService::Encoders::Compressed::Gzip.encode(json)
         base64_encoded = NewRelic::Base64.strict_encode64(gzipped)
         array = [PAYLOAD_VERSION, LAMBDA_MARKER, metadata, base64_encoded]
         ::JSON.dump(array)
       end
 
+      def determine_api_gateway_version
+        return unless @event
+
+        version = @event.fetch('version', '')
+        if version.start_with?('2.')
+          return 2
+        elsif version.start_with?('1.')
+          return 1
+        end
+
+        headers = headers_from_event
+        return unless headers
+
+        if @event.dig('requestContext', 'http', 'path') && @event.dig('requestContext', 'http', 'method')
+          2
+        elsif @event.fetch('path', nil) && @event.fetch('httpMethod', nil)
+          1
+        end
+      end
+
+      def process_api_gateway_info
+        api_v = determine_api_gateway_version
+        return unless api_v
+
+        info = api_v == 2 ? info_for_api_gateway_v2 : info_for_api_gateway_v1
+        info[:query_parameters] = @event.fetch('queryStringParameters', nil)
+
+        @http_method = info[:method]
+        @http_uri = http_uri(info)
+      end
+
+      def http_uri(info)
+        return unless info[:host] && info[:path]
+
+        url_str = "https://#{info[:host]}:#{info[:port]}#{info[:path]}"
+        if info[:query_parameters]
+          qp = info[:query_parameters].map { |k, v| "#{k}=#{v}" }.join('&')
+          url_str += "?#{qp}"
+        end
+
+        URI.parse(url_str)
+      rescue StandardError
+      end
```

Comment:
Why rescue here, but not elsewhere?

Reply:
Whoops! Thank you for spotting this. This `rescue` was only added for some debugging and I missed removing it. Removed with [c27f24c](https://github.com/newrelic/newrelic-ruby-agent/pull/2811/commits/c27f24c3d82aa2ddd10376fa224765a68234da98).